File: 3429-special-array-i/3429-special-array-i.py
class Solution(object):
    def isArraySpecial(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        n=int(len(nums))
        if n==1 :
            return True
        for i in range(n-1):
            if nums[i]%2 == nums[i+1]%2 :
                return False
        return True

File: 3429-special-array-i/3429-special-array-i.java
class Solution {
    public boolean isArraySpecial(int[] nums) {
        int n=nums.length,i;
        if(n==1)
        {
            return true;
        }
        for(i=0;i<n-1;i++)
        {
            if((nums[i]%2)==(nums[i+1]%2))
            {
                return false;
            }
        }
        return true;
    }
}